spec/helpers/application_helper_spec.rb

```diff
@@ -48,24 +48,29 @@
     end
   end
 
-  describe '#current_user_tracks' do
+  describe '#current_user_public_tracks' do
     context 'current_user is not present' do
-      specify { expect(helper.current_user_tracks).to be_nil }
+      specify { expect(helper.current_user_public_tracks).to be_nil }
     end
 
     context 'current_user is present' do
       let(:client) { double 'soundcloud client'}
       let(:current_user) { JSON.parse(File.read('spec/fixtures/eric.json')) }
-      let(:track)  { JSON.parse(File.read('spec/fixtures/coffee-machine-1.json')) }
+      let(:track)  { SoundCloud::HashResponseWrapper.new( JSON.parse(File.read('spec/fixtures/coffee-machine-1.json'))) }
+      let(:private_track) { SoundCloud::HashResponseWrapper.new( JSON.parse(File.read('spec/fixtures/coffee-machine-private.json'))) }
```

Comment:
@nynnejc @Powerglove @aydamacink here's a little refresher/additional explanation about this response wrapper. 

It looks like the SoundCloud::HashResponseWrapper is created [here](https://github.com/soundcloud/soundcloud-ruby/blob/8a4d809e9875cf21154fa58579692034546bc778/lib/soundcloud/hash_response_wrapper.rb), and it inherits from the Hashie::Mash class, which is explained [here](http://www.rubydoc.info/github/intridea/hashie/Hashie/Mash). 

None of that really matters, the important thing is is allows you to treat hashes differently. For example, normally, when accessing the value of a hash, you do the following:
```
hash = {}
hash{:a} = 'b'
hash[:a] => 'b' **normally you access the value like this**
```
The SoundCloud::HashResponseWrapper lets you instead do the following:
```
hash = {}
hash{:a} = 'b'
hash.a => 'b' **with the response wrapper, you can access the value like this!**
```

Wrapping our parsed JSON in the hash response wrapper is necessary in this test because [here](https://github.com/sounddrop/sounddrop/blob/1965d4b07a2d18925399b2e411d7836b097a8a7a/app/helpers/application_helper.rb#L23) we access the 'sharing' key in the track api response by using dot notation. Instead of doing `client.get('/me/tracks').select{|t| t['sharing'] == 'public'}` , we do `client.get('/me/tracks').select{|t| t.sharing == 'public'}`. However, the JSON fixture in our spec doesn't know about this notation, it thinks it's a regular hash & doesn't know about dot notation. So it complaints that it doesn't know how to respond to this & test tests fail. That's why we have to wrap it in the hash response wrapper. 

However, this whole thing could have been avoided if we just accessed the sharing key via the normal hash way ( `h[:a]` ). But since the soundcloud api allows us to access it differently, might as well do it. 

Make sense?